tests/core/crypto/test_crypto.h

```diff
@@ -68,6 +70,41 @@ TEST_CASE("[Crypto] PackedByteArray constant time compare") {
 	equal = crypto.constant_time_compare(p1, p2);
 	CHECK(!equal);
 }
-} // namespace TestCrypto
+
+TEST_CASE("[Crypto] Load and Save Key") {
+	// Create an instance of CryptoKey
+	Ref<CryptoKey> key = CryptoKey::create();
+
+	// Load the key from file
+	Error load_error = key->load(TestUtils::get_data_path("crypto/test_key.key"), false);
+
+	// Check if the key loaded successfully
+	CHECK(load_error == OK);
+
+	// Verify that the loaded key looks like a key
+	String key_data = key->save_to_string();
+	CHECK(key_data.begins_with("-----BEGIN RSA PRIVATE KEY-----"));
+
+	// Save the key to another temporary file
+	Error save_error = key->save("saved_key.key", false);
+
+	// Check if the key saved successfully
+	CHECK(save_error == OK);
+
+	// Load the key from the saved file
+	Ref<CryptoKey> saved_key = CryptoKey::create();
+	Error load_saved_error = saved_key->load("saved_key.key", false);
+
+	// Check if the saved key loaded successfully
+	CHECK(load_saved_error == OK);
+
+	// Verify that the re-loaded key also looks like a key
+	String saved_key_data = saved_key->save_to_string();
+	CHECK(saved_key_data.begins_with("-----BEGIN RSA PRIVATE KEY-----"));
+
+	// Cleanup temp file
+	remove("saved_key.key");
```

Comment:
This is from stdio - not sure how I have access to it here. It takes in a `char *` however, not the Godot String. Is there a better method to use for deleting the file?

Reply:
We don't use the standard library, and I think here we don't need to delete the file, we don't in several other tests such as `core/io/test_resource.h`

If we should delete you should use `DirAccess::remove`

Reply:
Since we moved the path this file is saved to in a previous change you suggested in this PR, I agree that we don't need to manually delete the file. Fixed that.